Adding provisioning and server db config docs (#2322)
kubeshop/tracetest: docs/docs/configuration/server.md

> # Configuring the Tracetest server

Reviewer: ```suggestion
# Configuring the Tracetest Server
```

> Tracetest requires a very minimal configuration to be launched, needing just the connection information to connect with the PostgreSQL database which is installed as part of the server install. There are a couple ways to provide this database connection information.

Reviewer: ```suggestion
Tracetest requires a very minimal configuration to be launched, needing just the connection information to connect with the PostgreSQL database which is installed as part of the server install. There are a couple of ways to provide this database connection information.
```

> For docker based installs, the server configuration file is placed in the ./tracetest/tracetest.yaml file by default when you run the 'tracetest server install' command and select the 'Using Docker Compose' option. The configuration file is mounted to `/app/config.yaml` within the Tracetest Docker container. When tracetest is run with a 'docker compose -f tracetest/docker-compose.yaml  up -d' command, the server will use the contents of this file to connect to the Postgres database. All other configuration data is stored in the Postgres instance.

Reviewer: ```suggestion
For Docker-based installs, the server configuration file is placed in the ./tracetest/tracetest.yaml file by default when you run the 'tracetest server install' command and select the 'Using Docker Compose' option. The configuration file is mounted to `/app/config.yaml` within the Tracetest Docker container. When Tracetest is run with a 'docker compose -f tracetest/docker-compose.yaml  up -d' command, the server will use the contents of this file to connect to the Postgres database. All other configuration data is stored in the Postgres instance.
```

> For Docker-based installs, the server configuration file is placed in the ./tracetest/tracetest.yaml file by default when you run the 'tracetest server install' command and select the 'Using Docker Compose' option. The configuration file is mounted to `/app/config.yaml` within the Tracetest Docker container. When Tracetest is run with a 'docker compose -f tracetest/docker-compose.yaml  up -d' command, the server will use the contents of this file to connect to the Postgres database. All other configuration data is stored in the Postgres instance.

Reviewer 2: ```suggestion
For Docker-based installs, the server configuration file is placed in the `./tracetest/tracetest.yaml` file by default when you run the `tracetest server install` command and select the `Using Docker Compose` option. The configuration file is mounted to `/app/config.yaml` within the Tracetest Docker container. When Tracetest is run with a `docker compose -f tracetest/docker-compose.yaml  up -d` command, the server will use the contents of this file to connect to the Postgres database. All other configuration data is stored in the Postgres instance.
```

> This is an example of a tracetest.yaml file:

Reviewer 2: ```suggestion
This is an example of a `tracetest.yaml` file:
```

```yaml
postgres:
  host: postgres
  user: postgres
  password: postgres
  port: 5432
  dbname: postgres
  params: sslmode=disable
```

> Alternatively, we support setting ana series of environment variables which can contain the connection information for the Postgres instance. If these environment variables are set, they will be used by the Tracetest server to connect to the database. The variables are:

Reviewer: ```suggestion
Alternatively, we support setting a series of environment variables which can contain the connection information for the Postgres instance. If these environment variables are set, they will be used by the Tracetest server to connect to the database.
```

> The list of environment variables, and example values is:

Reviewer: ```suggestion
The list of environment variables and example values is:
```

- TRACETEST_POSTGRES_HOST - example: postgres

> - TRACETEST_POSTGRES_PORT - example: 5432

Reviewer 2: ```suggestion
- `TRACETEST_POSTGRES_PORT: "5432"`
```

- TRACETEST_POSTGRES_DBNAME - example: postgres

> - TRACETEST_POSTGRES_USER - example: postgres

Reviewer 2: ```suggestion
- `TRACETEST_POSTGRES_USER: "postgres"`
```

> - `TRACETEST_POSTGRES_PASSWORD: "postgres"`

Reviewer 3: We also support `TRACETEST_POSTGRES_PARAMS`
